centos2ol.sh

```diff
@@ -51,127 +109,139 @@ have_program() {
-Your repositories have been restored to your previous configuration."
+error_handler1() {
+   abend $LINENO "Error condition intercepted. Could not install Oracle Linux packages."
 }
 
-generate_rpms_info() {
-    echo "Creating a list of RPMs installed $1 the switch"
-    rpm -qa --qf "%{NAME}|%{VERSION}|%{RELEASE}|%{INSTALLTIME}|%{VENDOR}|%{BUILDTIME}|%{BUILDHOST}|%{SOURCERPM}|%{LICENSE}|%{PACKAGER}\n" | sort > "/var/tmp/$(hostname)-rpms-list-$1.log"
-    echo "Verifying RPMs installed $1 the switch against RPM database"
-    rpm -Va | sort -k3 > "/var/tmp/$(hostname)-rpms-verified-$1.log"
+error_handler2(){
+   abend $LINENO "Error during execution of yum shell commands."
+}
+function data_dump
+{
+   if (( debug )); then 
+      echo "=== DATA DUMP ==="
+      cat $BASE/data_dump
+      echo "================="
+   fi      
 }
 
-## Start of script
+## Start of the script
+if (( $(id -u) > 0 )); then
+   abend $LINENO "You must run this script as root. You can try running sudo $0  or su -c $0 , if you want to start it from a regular account"
+fi
 
+if [[ -d $BASE ]]; then 
+   info $LINENO "ATTENTION: the directory BASE exists, so we assume that you are re-rerunning the conversion after an error"
+   if [[ -f $BASE/data_dump ]]; then 
+      if [[ ! -f $BASE/laststep ]]; then 
+         abend $LINENO "File $BASE/laststep does not exists"
+      fi
+      . $BASE/data_dump
+      STARTNO=`cat $BASE/laststep`
+      info $LINENO "ATTENTION: Attempting to continue from step $STARNO. If you want a different step please correct $BASE/laststep and re-run the script" 
+      echo "Correct execution is not garanteered. You can cancel the script within the next 10 sec "
+      sleep 10
+   else
+      mkdir -p BASE
+      if [[ ! -d $BASE ]]; then 
+         abend $LINENO "Can't create the directory $BASE"
+      fi   
+   fi  
+fi
+  
 reinstall_all_rpms=false
 
-verify_all_rpms=false
-
-while getopts "hrV" option; do
+while getopts "h:r" option; do
     case "$option" in
         h) usage ;;
+        d) debug=1;;
         r) reinstall_all_rpms=true ;;
-        V) verify_all_rpms=true ;;
         *) usage ;;
     esac
 done
 
-if [ "$(id -u)" -ne 0 ]; then
-    exit_message "You must run this script as root.
-Try running 'su -c ${0}'."
-fi
-
-echo "Checking for required packages..."
-for pkg in rpm yum curl; do
-    dep_check "${pkg}"
-done
-
-echo "Checking your distribution..."
-if ! old_release=$(rpm -q --whatprovides /etc/redhat-release); then
-    exit_message "You appear to be running an unsupported distribution."
-fi
-
-if [ "$(echo "${old_release}" | wc -l)" -ne 1 ]; then
-    exit_message "Could not determine your distribution because multiple
-packages are providing redhat-release:
-$old_release
-"
-fi
-
-# Collect information about RPMs before the switch
-if "${verify_all_rpms}"; then
-    generate_rpms_info before
+step_info $LINENO "Checking for required packages..."
+if (( STEPNO >= STARTNO  )) ; then
+   for pkg in rpm yum curl; do
```

Comment:
CentOS 8 to Oracle Linux 8 requires `dnf` not `yum` and `rpm` and `curl` are mandatory on any instance. If `rpm` or `curl` are missing, it would not be possible to install them.